packages/code-studio/src/main/AppMainContainer.jsx

```diff
@@ -225,6 +227,10 @@ export class AppMainContainer extends Component {
     this.setState({ isSettingsMenuShown: true });
   }
 
+  handleShortcutUpdate() {
+    this.emitLayoutEvent(NotebookEvent.UPDATE_SHORTCUTS);
```

Comment:
As I mentioned on our call... I think the event should be emitted through the `Shortcut` or `ShortcutRegistry` itself. It seems logical that any consumer of Shortcuts in the @deephaven/components project may want to listen to changes whenever they occur to shortcuts. So something to trigger in the Shortcut.setKeyState I think, and ability to add/remove listeners.

Reply:
I currently made it to where the changes will emit an update event from the registry w/ an array of changed shortcuts. Let me know if I should switch it to be emitted from the shortcut class. I thought that might be a bit more annoying since then there's an event for each shortcut and you could just listen to the registry event and filter for any shortcuts you care about.

Could also have each shortcut's update method emit an event to the registry to then emit an "any shortcut changed" event?